### PR TITLE
PHOENIX-6829: Add code coverage report aggregation with Jacoco

### PR DESCRIPTION
**Pull request by horvathdora**

### CODE REVIEW

File: phantomjsdriver.log [hunk 1]
```diff
@@ -0,0 +1,675 @@
+[INFO  - 2022-08-22T07:28:26.132Z] GhostDriver - Main - running on port 6361
```
Comment (stoty) [inline phantomjsdriver.log:1]: This seems to be an unintended addition.

File: test-reporting/pom.xml [hunk 2]
```diff
@@ -0,0 +1,108 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (stoty) [inline test-reporting/pom.xml:1]: I'd prefer if we could do this without an additional module.
Maybe this could be done in the phoenix-assembly module ?

**Comment by stoty** [comment]
The JavaDoc Yetus check fails.
Are you able to generate the javadocs with the patch, @horvathdora  ?
`mvn --batch-mode -Dmaven.repo.local=/home/jenkins/jenkins-home/workspace/enix-PreCommit-GitHub-PR_PR-1524/yetus-m2/phoenix-master-patch-0 --threads=1 -DPhoenixPatchProcess -Dskip.code-coverage clean javadoc:javadoc -DskipTests=true`

**Comment by horvathdora** [comment]
@stoty No, I can't generate the javadocs report neither with the patch nor on the master branch.

**Comment by stoty** [comment]
That's strange.
I've just run `mvn --batch-mode --threads=1 -DPhoenixPatchProcess -Dskip.code-coverage clean javadoc:javadoc -DskipTests=true` on both master and your patch without errors.

**Review by stoty — APPROVED** [review]
+1 LGTM

**Comment by stoty** [comment]
Before committing, I want to check if the Javadoc yetus errors are related to the patch.

File: pom.xml [hunk 3]
```diff
@@ -148,7 +148,7 @@
     <spotbugs-maven-plugin.version>4.4.1</spotbugs-maven-plugin.version>
     <spotbugs.version>4.4.1</spotbugs.version>
     <jacoco-maven-plugin.version>0.8.8</jacoco-maven-plugin.version>
-    <sonar-maven-plugin.version>3.9.1.2184</sonar-maven-plugin.version>
+    <sonar-maven-plugin.version>3.6.0.1398</sonar-maven-plugin.version>
```
Comment (mnpoonia) [inline pom.xml:151]: Why are we downgrading the version? Can't we keep the current version as it is?

**Comment by stoty** [comment]
Please rebase the patch @horvathdora , Yetus complains that it cannot apply it on top of master.

**Comment by stoty** [comment]
restarted the CI job manually

**Comment by stoty** [comment]
The last run looks good, there is only a single unrelated IT failure.

**Comment by stoty** [comment]
But then again, there is no actual change in this PR now.
Something is not right, @horvathdora.

**Comment by horvathdora** [comment]
Change was merged to master branch.

**Pull request closed without merging**